### PR TITLE
doc: Add release note for 0.22.0

### PR DESCRIPTION
**Pull request by mumoshu**
As it turned out to be the biggest release ever, I was afraid I might not be able to write a summary of changes that communicates well. Here is my attempt. Please review and leave any comments so that we can be more confident in this release. Thank you!

### CODE REVIEW

**Comment by iniinikoski** [comment]
> As it turned out to be the biggest release ever, I was afraid I might not be able to write a summary of changes that communicates well. Here is my attempt. Please review and leave any comments so that we can be more confident in this release. Thank you!

I'm still quite newbie with ARC - but the notes looked very clear at least to me... :)

File: docs/releasenotes/0.22.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# actions-runner-controller v0.22.0
+
+This version of ARC focuses on scalability and reliablity of runners.
+
+## GitHub API Cache
+
+In terms of scalability, ARC now caches GitHub API responses according to their recommendation(=Cache-Control header).
```
Comment (TingluoHuang) [inline docs/releasenotes/0.22.md:7]: do we want to link to the doc https://docs.github.com/en/rest/overview/resources-in-the-rest-api#conditional-requests?
ex: [Cache-Control header](https://docs.github.com/en/rest/overview/resources-in-the-rest-api#conditional-requests)

File: docs/releasenotes/0.22.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+As long as GitHub keeps its current behavior, it will result in ARC to cache various List Runners API and List Workflow Jobs calls for 60 seconds.
+
+The cache for List Runners API is expecially important, as their responses can be shared between every runner under the same scope (repository, organization, or enterprise).
+
+In previous versions of ARC, the number of List Runners API calls had scaled proportional to the number of runners managed by ARC.
+Thanks to the addition of cache, since v0.22.0, it may scale proportional to the number of runner scopes (=The number of repositories for your repository runners + The number of organizations for your organizational runners + The number of enterprises for your enterprises). You might be able to scale to hundreds of runners depending on your environemnt.
```
Comment (TingluoHuang) [inline docs/releasenotes/0.22.md:13]: maybe with `+ The number of enterprises for your enterprise runners`. 😄

File: docs/releasenotes/0.22.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+Please share your experience if you successfully scaled to a level that wasn't possible with previous versions!
+
+## Improved Runner Scale Down Process
+
+In terms of reliability, the first thing to note is that it has a new scale down process for both RunnerDeployment and RunnerSet.
+
+Previously every runner pod can restart immediately after the completion, while ARC might mark the same runner pod for deletion due to scale down. That resulted in
```
Comment (TingluoHuang) [inline docs/releasenotes/0.22.md:21]: do we want to link to the issue for more context if anyone interested?

File: docs/releasenotes/0.22.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+various race conditions that resulted in the runner get terminated prematurely while running a workflow job.
+
+And it's now fixed. The new scale down process ensures that the runner has been registered successfully and then de-registered from GitHub Actions, before starting the runner pod deletion process.
+Any runner pod can't be terminated while being restarting or running a job now, which makes it impossible to be in the middle of running a workflow job when a runner pod is being terminated. No more race conditions.
+
+## Prevention of Unnecessary Runner Pod Recreations
+
+Another reliability enhancement is based on the addition of a new field, `EffectiveTime`, to our RunnerDeployment and RunnerSet specifications.
```
Comment (TingluoHuang) [inline docs/releasenotes/0.22.md:29]: do we need to highlight what scale mode the `EffectiveTime` is good for? ex: webhook-based scale vs runner busy-based.

File: docs/releasenotes/0.22.md (hunk 1 above)
Comment (mumoshu) [reply docs/releasenotes/0.22.md]: Thanks! Added in [e402b2e](https://github.com/actions-runner-controller/actions-runner-controller/pull/1199/commits/e402b2e8fd0e32598e326dbad8a3cb3d66401a5e)

File: docs/releasenotes/0.22.md (hunk 2 above)
Comment (mumoshu) [reply docs/releasenotes/0.22.md]: Good catch 👍 Addressed in [e402b2e](https://github.com/actions-runner-controller/actions-runner-controller/pull/1199/commits/e402b2e8fd0e32598e326dbad8a3cb3d66401a5e)

File: docs/releasenotes/0.22.md (hunk 3 above)
Comment (mumoshu) [reply docs/releasenotes/0.22.md]: I think so! [e402b2e](https://github.com/actions-runner-controller/actions-runner-controller/pull/1199/commits/e402b2e8fd0e32598e326dbad8a3cb3d66401a5e)

File: docs/releasenotes/0.22.md (hunk 4 above)
Comment (mumoshu) [reply docs/releasenotes/0.22.md:33]: It's enabled for both modes and there's no way to disable it now. I tried to make it clear in 0f3ead1. Would you mind confirming? 😄

**Review by TingluoHuang — APPROVED** [review]

**Comment by mumoshu** [comment]
Probably https://github.com/actions-runner-controller/actions-runner-controller/pull/1204 worth covered too. It has a good side-effect of mostly skipping RemoveRunner API calls for ephemeral runners, which in combination with the addition of GitHub API cache, provides us more scalability.

**Comment by mumoshu** [comment]
TODO: https://github.com/actions-runner-controller/actions-runner-controller/issues/1189 needs to be included once we enabled the `RUNNER_FEATURE_FLAG_EPHEMERAL` by default via a controller change.

**Comment by mumoshu** [comment]
Added [9acba10](https://github.com/actions-runner-controller/actions-runner-controller/pull/1199/commits/9acba103fea98bbe7d7f6945e3baf4f35e945622) to cover #1204 and #1189(PR #1211)

**Pull request merged**